### ISSUE TITLE
[📚] AdMob documentation availability

### ISSUE THREAD

**Issue by mmskvortsov** [invertase/react-native-firebase#5329]
## AdMob documentation availability

Hi there! I've noticed that AdMob documentation is not available on the official website: https://rnfirebase.io/admob
There is a 404 error:
<img width="626" alt="image" src="https://user-images.githubusercontent.com/3917463/118607859-bd740300-b7c1-11eb-9718-88f242f164a0.png">

Is there any issue with that or this part of documentation turned off intentionally? 
Thanks!

**Comment by Deodes** [comment]
+

**Comment by mikehardy** [comment]
This came up as a comment on an unrelated issue and I think I may understand the confusion now - pasting in my comment from there:

Ah - sorry, we've already merged the change where Admob is removed from firebase following it's removal upstream, and the documentation site goes live immediately after merges to main branch.

It is not as great of an experience but for now until we have the admob code re-hosted in a non-firebase repo you can go direct to the usage docs and types / API reference from the v11.5.0 (current stable) tag

https://github.com/invertase/react-native-firebase/tree/v11.5.0/docs/admob
https://github.com/invertase/react-native-firebase/blob/v11.5.0/packages/admob/lib/index.d.ts

Normally it's a great thing that the docs site is updated immediately between releases, but in this case it means the documentation for a module that is still live (and will always be, in that version) is gone. I'll have to think on that to see if there's some better way we can keep docs for old versions alive as they are still quite useful

**Comment by mikehardy** [comment]
I've pinned this issue for others

**Comment by mmskvortsov** [comment]
@mikehardy Thanks for clarification!

**Comment by Kamani** [comment]
@mikehardy Sorry it may seem stupid question but please help me.
You said Admob is removed. does it mean that we will be not able to use yarn add @react-native-firebase/admob? As document is not available does it mean that?

**Comment by mikehardy** [comment]
@Kamani v11.5.0 contains admob. That is the current latest release so you may use the command you mention.
As admob will *not* be in the next release, it may be best to specify the version precisely so it does not auto-update other packages, all of your react-native-firebase packages should be pinned to exactly '@11.5.0' vs having any semver range

**Comment by Kamani** [comment]
@mikehardy Thank you for your reply. I really appriciate. Admob will be again integrade in future release? Or you guys are planning to remove it permenantly?

**Comment by mikehardy** [comment]
It's gone from upstream. The Google Firebase team have removed it. We track them. It will never be in react-native-firebase again.

We have plans to re-host the code on an external repository, but you will not be able to update react-native-firebase and use admob at the same time until we do so

This may not be great to hear but please understand we are downstream consumers of the SDKs. This is not our decision.

**Comment by Kamani** [comment]
@mikehardy I can understand. Thank you for explaining in detail. May be this detailed comment will be also useful for new users.

**Comment by mikehardy** [comment]
V11.5 was just fine 2 days ago and is still fine today? Please be a little patient and the code will be in a new repo. Or there is react-native-admob I think?

**Comment by inferusvv** [comment]
> V11.5 was just fine 2 days ago and is still fine today? Please be a little patient and the code will be in a new repo. Or there is react-native-admob I think?

Oh, if you are going to create separate library then it is cool.

`react-native-admob` last updated 2 years ago and I had problems while installing on 0.64 RN

**Comment by ifsnow** [comment]
@mikehardy Thank you for your hard work and responses.

Do you have an expected timeline for `@invertase/react-native-admob`? I'm worried that I will not be able to apply the new version of `react-native-firebase` for a long time. There is no problem at all right now :)

**Comment by mikehardy** [comment]
Unfortunately not, and with admob it's a little tricky because even before the most recent change the code here was already in need of a breaking forward port #5150 - that forward port itself means that step 1 is simply to move the code, step 2 is then to forward port it (otherwise there will be collisions for folks trying to use up to date GoogleAppMeasurement symbols - which Firebase Analytics shares).

This is all unfunded work so it's a lower priority - I do love making things work but I have to prioritize as there is of course an infinite task queue ;-).

If you wanted to help it would be really really welcome and even before moving (step 1 as I listed it) it is actually possible to start with what I listed as step 2 by branching from the v11.5.0 tag here and beginning the forward port based on google documentation. Any work done here can then be merged onto a branch and I'll just use the branch for the repo extraction, but the forward port work will have the support of all our E2E testing / test app infrastructure which is tremendously helpful I think.

**Comment by mikehardy** [comment]
That offer is open to anyone of course! Any future admob repo will have to have a large community participation to move forward, but the test infrastructure and CI and build/release tooling and stuff - that's what we / the future repo will get all for free so it's easy to contribute I hope.

**Comment by aleluiah** [comment]
setDebugGeography dosen't reflect even when i whitelist my deviceId

**Comment by spazfeed** [comment]
Use   [react-native-admob-next](https://github.com/izetmolla/react-native-admob-next) for admob this is based on expo-ads-admob.

**Comment by mikehardy** [comment]
That sounds good and could be a good solution @spazfeed do you know if it's already forward-ported to the new AdMob breaking changes? I'm just curious, because we are not forward-ported here, and if that library already is then it is ahead of anything we'd have by splitting the previous code here out

**Comment by tominou** [comment]
The best alternative is Admob Plus (https://admob-plus.github.io/docs/react-native)
It is up to date, I used it for years with Cordova and now with React native.

**Comment by mars-lan** [comment]
> That sounds good and could be a good solution @spazfeed do you know if it's already forward-ported to the new AdMob breaking changes? I'm just curious, because we are not forward-ported here, and if that library already is then it is ahead of anything we'd have by splitting the previous code here out

Most probably not. It's still on [v19 (Android)](https://github.com/izetmolla/react-native-admob-next/blob/main/android/build.gradle#L55) & [v7 (iOS)](https://github.com/izetmolla/react-native-admob-next/blob/main/react-native-admob-next.podspec#L23).

**Comment by nonth** [comment]
> > That sounds good and could be a good solution @spazfeed do you know if it's already forward-ported to the new AdMob breaking changes? I'm just curious, because we are not forward-ported here, and if that library already is then it is ahead of anything we'd have by splitting the previous code here out
> 
> Most probably not. It's still on [v19 (Android)](https://github.com/izetmolla/react-native-admob-next/blob/main/android/build.gradle#L55) & [v7 (iOS)](https://github.com/izetmolla/react-native-admob-next/blob/main/react-native-admob-next.podspec#L23).

I just create pull request for update Mobile Ads SDK [here](https://github.com/izetmolla/react-native-admob-next/pull/1) or you can use it directly from [my fork](https://github.com/nonth/react-native-admob-next)

**Comment by Mikenso** [comment]
It seems like @invertase/react-native-admob has huge advantage (comparing to react-native-admob-next)  as it allows to create few interstitial instances and react-native-admob-next does not (correct me if I am wrong)
So I hope @invertase/react-native-admob will find itself in good hands of developer

**Comment by mikehardy** [comment]
Still on the radar as something to port out yes but I was curious yes if it still had value compared to any other efforts, thanks for chiming in

**Comment by seamive** [comment]
Any plans to update AdMob & IID to the latest version? Stuck on 11.5.0 due to both.

**Comment by maxcda101** [comment]
I hope you develop admob-next soon. Currently all the admob react native libraries I see have errors, or are not updated to the latest GMA. Thank you

**Comment by mikehardy** [comment]
I'm going to re-open this issue as it is where I/we point most people, and it should be clear that admob as a separate repo is still an active item in our backlog. I will issue announcements here as I have them.

**Comment by sdb1228** [comment]
Hey @mikehardy I assume most people make money off of this integration/package existing. Is there a price tag on getting this to work/migrated? I personally wouldn’t mind donating money towards this specific feature given its longevity is crucial for my future apps viability. 

I could also dedicate time obviously as well to get this to work but I assure you it would take me triple the time it might take someone else who is more familiar with the underlying tech and I would be more than happy to pay someone with more experience to do it.

**Comment by tarouboy** [comment]
Agree with @sdb1228 
I wouldn't mind paying for a well maintained package. How about making it a paid package like Notifee?

**Comment by inferusvv** [comment]
For myself I found https://github.com/appodeal/react-native-appodeal. It is not only AdMob, but it works fine on latest RN

**Comment by tarouboy** [comment]
> For myself I found https://github.com/appodeal/react-native-appodeal. It is not only AdMob, but it works fine on latest RN

Appodeal is a good alt but I have concern about how they use and collect all the data. :/

**Comment by Acetyld** [comment]
> For myself I found https://github.com/appodeal/react-native-appodeal. It is not only AdMob, but it works fine on latest RN

Do you got a demo project perhaps, setting it up currently, did you do the admob adapter etcc as wel? Because the RN documentantion is pretty minimal or maybe @tarouboy has a github repo?

**Comment by inferusvv** [comment]
> > For myself I found https://github.com/appodeal/react-native-appodeal. It is not only AdMob, but it works fine on latest RN
> 
> Do you got a demo project perhaps, setting it up currently, did you do the admob adapter etcc as wel? Because the RN documentantion is pretty minimal or maybe @tarouboy has a github repo?

I can't check admob now, because I am banned for 30 days for now, but actually you should enable AdMob in Appodeal dashboard and thats it. Nothing from app code side.

**Comment by mikehardy** [comment]
For my part, I wish I just had a shiny new git URL to point you all at :-). It is still on my general priority list but sadly just has not hit the #1 spot yet relative to priorities that have a direct obligation (that is: are directly related to Firebase and Firebase APIs). I have surfaced this to the Invertase folks tho, perhaps there is potential for a Notifee-like library that's free to test-integrate and has a one-time license etc but then has a funding model for sustained / higher-priority effort.

**Comment by khalisafkari** [comment]
I use the react native firebase plugin but I don't use the admob plugin because I write using tapdaq.com because it's easier to manage ads and supports mediatation and fbads bidding

if you like you can try to see my repo https://github.com/khalisafkari/react-native-tapdaq-ad

but I haven't written the installation document and api but you can see the api in the src folder

**Comment by birdofpreyru** [comment]
These attempts to sell 3rd party ads-related products are annoying. The time it takes Invertase to update and split the library is also annoying, considering quite small size of the package in question. I'm considering to do my own fork with blackjack and hookers 🤖

**Comment by wjaykim** [comment]
Check out my new package https://github.com/react-native-admob/admob

**Comment by mars-lan** [comment]
> Check out my new package https://github.com/react-native-admob/admob, which supports new SDK versions and newly added hooks for Interstitial and Rewarded Ad.

@wjaykim  Thanks fo this great library. Please see my comments in https://github.com/react-native-admob/admob/issues/2

**Comment by mikehardy** [comment]
@wjaykim - looks cool - I notice in the install instructions there is no version of admob specified - what's the version compatibility like? Is it all forward-ported to the new versions of AdMob?

**Comment by wjaykim** [comment]
> @wjaykim - looks cool - I notice in the install instructions there is no version of admob specified - what's the version compatibility like? Is it all forward-ported to the new versions of AdMob (and thus compatible with analytics / google app measurement pulled in from here)?

I believe for now it's compatible for all Google Mobile Ads SDK version 8+ for iOS and 20+ for Android, but I have not tested with all of them.

**Comment by mars-lan** [comment]
> I believe for now it's compatible for all Google Mobile Ads SDK version 8+ for iOS and 20+ for Android, but I have not tested with all of them.

In that case you may wish to add the valid version range in Podfile & build.gradle.

**Comment by birdofpreyru** [comment]
Not sure I support the enthusiasm over the new package by @wjaykim. At a brief look it does not seem to be a fork of this (Invertase's) library, which IMHO was the best RN AdMob library so far, and it uses a different API, thus not a simple drop-in replacement. Not sure it supports ads personalization opt-in/-out, as this lib does. Also other comments pointing out it was not packaged and distributed as per best practices do not increase my trust in it :(

**Comment by unstableair** [comment]
At least it's in active development (3 commits today, 3 yesterday), @wjaykim seems responsive to feedback and accepts PRs
It's the only RN admob solution I can find that supports the latest major release of the admob SDK
Regarding personalisation, for some time the Invertase library was broken and ONLY showed non-personalised ads (https://github.com/invertase/react-native-firebase/pull/5262), so no-one always gets it right first time.
It might not be production ready (I haven't tried it yet) but the dev deserves support and encouragement.

**Comment by birdofpreyru** [comment]
> _no-one always gets it right first time_

Exactly, and this is a general wisdom, that's why my criticism: it would be a way better to build upon Invernase lib, rather than building from scratch and re-inventing the wheel.

**Comment by wjaykim** [comment]
@birdofpreyru Yes, it is true. My package is under active development and many new issues are opening more than I expected. Also since the package is my first attempt to distribute to public, so the package may not follow best practices as other general packages does. So please understand those points 😅

**Comment by Mikenso** [comment]
So will '@invertase/react-native-admob' ever be officially port out as separate npm package, at least as it is at 11.5.0 version?

**Comment by huseyin39** [comment]
Guys, @react-native-firebase/admob@v11.5.0 is compatible with wich version of RN ?It's crashing with RN 0.64.

**Comment by Mikenso** [comment]
@huseyin39 It's crashing when you have latest gms sdk at your project. Doesn't matter if it is RN 0.64

**Comment by huseyin39** [comment]
@Mikenso which version of the gms sdk are we supposed to have?

**Comment by birdofpreyru** [comment]
@huseyin39 it works for me with v4.3.4

**Comment by Mikenso** [comment]
@huseyin39 for Android It is 19.8.0

**Comment by mikehardy** [comment]
v11.5.0 was issued while react-native 0.64 existed and should work fine with it
@Mikenso yes it will - it took a long long time to get the AppCheck module up but I just finished that work #5581 which was blocking *everything* since approximately May. A more rapid tempo will resume here now https://github.com/invertase/react-native-firebase/commits/master ...

**Comment by huseyin39** [comment]
@Mikenso Which SDK are you talking about? This one "implementation 'com.google.android.gms:play-services-ads:19.8.0'" on app/build.gradle  ?

I added it alogn with "classpath('com.google.gms:google-services:4.2.0')" on build.gradle and still crashing:

The error is *Invalid application ID.

Thank you.

**Comment by birdofpreyru** [comment]
@huseyin39 I have in app/build.gradle `classpath 'com.google.gms:google-services:4.3.4'`, and it works. I don't think I have `com.google.android.gms:play-services-ads` explicitly mentioned in gradle files.

**Comment by Mikenso** [comment]
@huseyin39 Try these 
classpath 'com.android.tools.build:gradle:4.2.2'
classpath 'com.google.gms:google-services:4.3.8

> The error is *Invalid application ID
Maybe you forgot to add application ID at root folder

**Comment by CyrusZei** [comment]
Wish I would have that deeper knowledge so that I could help out 😞

**Comment by huseyin39** [comment]
Ok guys, it seems working thanks I had indeed an issue with the ID and the version of gms!

**Comment by mikehardy** [comment]
We still have plans to shift our AdMob implementation from v11.5 out to a standalone and then update it and if you've seen the flurry of activity here recently (just released Installations module, App Distribution module, App Check module!) you'll see I've nearly cleared the backlog that built up while I was solving a nasty App Check issue.

I'm mostly writing to say that if you are still using react-native-firebaes v11.5 to access the AdMob APIs, there was an issue with react-native v0.65 that caused warnings in the log - #5614 but as I went to fix it with #5616 it did not seem like the log warnings were an actual problem. No, they aren't great but if you need the combination of react-native-firebase v11.5.0 and react-native v0.65+ it should be okay to try and then use, if it works for you.

**Comment by seamive** [comment]
@mikehardy when can we expect the Admob shift and what Admob package you recommend to use as standalone until then?

**Comment by mikehardy** [comment]
Soon. V11.5 here.

**Comment by CyrusZei** [comment]
Yeah, I am so sponsoring this!! Awesome job @mikehardy

**Comment by babyrusa** [comment]
@mikehardy  Hello, side question. Since I use ` @react-native-firebase/admob` , I don't see this in rnfirebase documentation but do I have to add `com.google.android.gms:play-services-ads` in my `app/build.gradle` file? I do have `classpath 'com.google.gms:google-services:` in my main build.gradle and the ads worked sometimes but Google kept limiting my ads so I'm not sure if that is because I'm missing  `com.google.android.gms:play-services-ads` in my `app/build.gradle` file
Thank you

**Comment by mikehardy** [comment]
Google may limit ads for any number of reasons, none of which we can help with unfortunately. If the ads worked at all, the software is working, that's the only knowable thing, as a module maintainer. Sorry I can't help more

**Comment by babyrusa** [comment]
@mikehardy I know they do this to a lot of people but my question is more on the implementation is if I have to add com.google.android.gms:play-services-ads to my app/build.gradle file if I'm already using `@react-native-firebase/admob` since it's not in rnfirebase doc but it's in Google Admob SDK doc

**Comment by mikehardy** [comment]
So you can do `. /gradlew :app:dependencies` (I think that is the right command) and save the output and see your whole android dependency tree. You can also see what we add to the build.gradle by just looking at it in node_modules right? Investigate, experiment, it's the best way

**Comment by babyrusa** [comment]
@mikehardy  thank you!

**Comment by dylancom** [comment]
@mikehardy would like to help with upgrading to the latest SDK and getting new features in such as "App Open Ads". Have been relying on this package for quite some time now.

**Comment by haresh4d** [comment]
> > > That sounds good and could be a good solution @spazfeed do you know if it's already forward-ported to the new AdMob breaking changes? I'm just curious, because we are not forward-ported here, and if that library already is then it is ahead of anything we'd have by splitting the previous code here out
> > 
> > 
> > Most probably not. It's still on [v19 (Android)](https://github.com/izetmolla/react-native-admob-next/blob/main/android/build.gradle#L55) & [v7 (iOS)](https://github.com/izetmolla/react-native-admob-next/blob/main/react-native-admob-next.podspec#L23).
> 
> I just create pull request for update Mobile Ads SDK [here](https://github.com/izetmolla/react-native-admob-next/pull/1) or you can use it directly from [my fork](https://github.com/nonth/react-native-admob-next)

How can I use your fork? How to install it?

**Comment by mikehardy** [comment]
Just FYI, we're getting close here, I'm at the "setting up e2e testing" part but the module from previous code here / extracted moved is public now https://github.com/invertase/react-native-google-ads/

next steps are finish a quick e2e bringup to verify functionality, then forward port

No documentation is available yet and it's not on npmjs.com but git references should work

for the migration the general idea is "admob" is now "googleAds", and the example app shows how to use `app.json` (not `firebase.json` to put in the config keys

It's my current project here, it will be finished before I move to something else

**Comment by haresh4d** [comment]
Found a great alternative which also supports app open ads. 

https://github.com/react-native-admob/admob

**Comment by bilarslan** [comment]
> Just FYI, we're getting close here, I'm at the "setting up e2e testing" part but the module from previous code here / extracted moved is public now https://github.com/invertase/react-native-google-ads/
> 
> next steps are finish a quick e2e bringup to verify functionality, then forward port
> 
> No documentation is available yet and it's not on npmjs.com but git references should work
> 
> for the migration the general idea is "admob" is now "googleAds" / "AdMob" in types is now "GoogleAds" etc, and the example app shows how to use `app.json` (not `firebase.json`) to put in the config keys
> 
> It's my current project here, it will be finished before I move to something else

Glad to hear that 🎉 Looking forward to first release 👀

**Comment by mikehardy** [comment]
When it's released it will hopefully support the entire API surface area of google ads, and one of the reasons to do a native module wrapper vs pure JS is that if it's possible with the SDK natively (app open ads, for example) it will be possible with the wrapper.

As for progress the last time I had a good chunk of hours to throw at it, I thought it was pretty close :shrug: - it is slow going though, I do understand. Only so many hours in the day and I have a couple hot work projects at the moment

**Comment by RodolfoGS** [comment]
@mikehardy I've a question regarding to: _one of the reasons to do a native module wrapper vs pure JS is..._

The previous library (`@react-native-firebase/admob`) was a "native module wrapper" or a "pure JS"? So, the future library will work with the same architecture or will be different?

Thanks

PD: Thank you so much for your hard work!

**Comment by mikehardy** [comment]
Hi @RodolfoGS - the previous incarnation of admob in `@react-native-firebase/admob` was a native module wrapper, yes. If you just want JS you can probably do that right now without waiting on anyone :-), but it won't have any native integrations

**Comment by Pingou** [comment]
@mikehardy Any update? I understand you must be very busy, I'm just wondering if I should wait out a bit more or switch to another package.

**Comment by dylancom** [comment]
I submitted a few PR's which makes the package usable (at least on iOS). For the time being (no npm package yet) it can be added to your projects package.json using: `"react-native-google-ads": "github:invertase/react-native-google-ads"`.

**Comment by mikehardy** [comment]
Happy to help! Thanks for helping get out the door. Sorry I did not have the time but I do want to make it clear that the goal is for the package to be alive, and I will look at and merge any PRs that help get it there, as I have with those from @dylancom :-)

**Comment by mikehardy** [comment]
The new repo is alive and updated to the latest SDKs for AdMob now. It's still moving very quickly and you should test carefully before deploying on it, but it's alive. I'm going to close this, cheers

**Comment by Ronald3217** [comment]
> This came up as a comment on an unrelated issue and I think I may understand the confusion now - pasting in my comment from there:
> 
> Ah - sorry, we've already merged the change where Admob is removed from firebase following it's removal upstream, and the documentation site goes live immediately after merges to main branch.
> 
> It is not as great of an experience but for now until we have the admob code re-hosted in a non-firebase repo you can go direct to the usage docs and types / API reference from the v11.5.0 (current stable) tag
> 
> https://github.com/invertase/react-native-firebase/tree/v11.5.0/docs/admob https://github.com/invertase/react-native-firebase/blob/v11.5.0/packages/admob/lib/index.d.ts
> 
> Normally it's a great thing that the docs site is updated immediately between releases, but in this case it means the documentation for a module that is still live (and will always be, in that version) is gone. I'll have to think on that to see if there's some better way we can keep docs for old versions alive as they are still quite useful


Do you have any idea if it is being updated, or are you planning to update the library in the future? version 11.5.0 is the latest stable version and it works perfectly, but I have doubts if updating the @react-native-firebase/app core to version 14.5.0 makes the application stop working.

Have you used admob v11.5.0 with app v14.5.0?
or is this not possible? Sorry for the question, it may sound a bit silly.

**Comment by mikehardy** [comment]
@Ronald3217 you are in luck! It was peeled out to https://github.com/invertase/react-native-google-mobile-ads and has gone through a few revisions so it is all up to date, it now appears to be stable and we're consolidating the rest of the react-native admob packages in to it (by working with / collaborating with their maintainers)

Go grab that package and move forward with admob, fully separate from Firebase

**Comment by mikehardy** [comment]
Oh, and as for versioning, never mix versions: https://invertase.io/blog/react-native-firebase-versioning

**Comment by Ronald3217** [comment]
> @Ronald3217 you are in luck! It was peeled out to https://github.com/invertase/react-native-google-mobile-ads and has gone through a few revisions so it is all up to date, it now appears to be stable and we're consolidating the rest of the react-native admob packages in to it (by working with / collaborating with their maintainers)
> 
> Go grab that package and move forward with admob, fully separate from Firebase

i am working with @react-native-firebase/app and @react-native-firebase/admob versions 11.5.0 (in released apps) but building one and using firestore and app v14.5.0.

From what I understand you no longer recommend working with the /admob module and instead use react-native-google-mobile-ads (because I've been using the /admob module for a long time and I'm used to its implementation) but it is better implement the new library, I'll have to learn how to use it.

**Comment by birdofpreyru** [comment]
@Ronald3217 https://github.com/invertase/react-native-google-mobile-ads, as of its **v4.2.0** is just `react-native-firebase/admob` module split into a standalone package. You won't have to learn how to use it, just carefully rename library imports and config files in your source code.

**Comment by Ronald3217** [comment]
> @Ronald3217 https://github.com/invertase/react-native-google-mobile-ads, as of its **v4.2.0** is just `react-native-firebase/admob` module split into a standalone package, with a bunch of updates / fixes done on the way. You won't have to learn how to use it, just carefully rename library imports and config files in your source code, and maybe do a few other minor changes according to the changelog.
> 
> RN Firebase 11.5.0 is almost one year old at this point, still using it you are probably missing some performance / features because of older underlying SDKs, and keep around some bugs fixed since then.

Reading the documentation, I don't see that dependencies need to be implemented in build.gradle app and project.
I just have to install it, add app.json and implement it as it says in the documentation?

Thanks in advance for taking the time to respond.

**Comment by birdofpreyru** [comment]
Yeah, just follow the docs. I believe, there is nothing to do in gradle config for this lib.

**Comment by Ronald3217** [comment]
@mikehardy thank you very much, have a nice afternoon.

**Comment by Ronald3217** [comment]
> Yeah, just follow the docs. I believe, there is nothing to do in gradle config for this lib.

Hello, I have a question for you, have you implemented the new library https://docs.page/invertase/react-native-google-mobile-ads? I've been trying all day but I couldn't.

I have sdk 29, do you think that is the problem. If so, which one should I use?

**Comment by mikehardy** [comment]
@Ronald3217 if you have a problem with that library, please post in that library. Prior to doing so, please examine the example app (and/or the example construction script - it's scripted: https://github.com/invertase/react-native-google-mobile-ads/blob/main/refresh-example.sh) to make sure you are following best practices and that your local environment is not busted in some way

**Comment by sdb1228** [comment]
I know this is closed but @mikehardy and Team over at invertase have done a fantastic job with the new admob package and made migrating from this package super simple (removed firebase admob from package.json cleared dependencies for ios and node_modules reinstalled and added my app id to my app.json) I personally strongly encourage anyone who hasn't yet migrated to do so. Thanks again to @mikehardy and team for putting for the effort to keep this package running.
